src/Symfony/Bundle/SecurityBundle/DependencyInjection/MainConfiguration.php

```diff
@@ -435,6 +435,10 @@ private function addPasswordHashersSection(ArrayNodeDefinition $rootNode): void
                         ->canBeUnset()
                         ->performNoDeepMerging()
                         ->beforeNormalization()->ifString()->then(fn ($v) => ['algorithm' => $v])->end()
+                        ->validate()
+                            ->ifTrue(fn ($v) => isset($v['migrate_from'], $v['id']) && 0 !== \count($v['migrate_from']))
+                            ->thenInvalid('You cannot use "migrate_from" when using a custom service id.')
```

Comment:
As it was silently ignored before, it would be better to trigger a deprecation than throwing an error, to avoid breaking projects in 6.4

Reply:
Since it would mean such projects are configured in a way the developers thinks it'll handle the password migration properly, but is not, making them aware of the issue the hard way might be better? 

The fix for removing this exception is straightforward enough, unless this config is hidden somehow in a dependency or whatever (but I would expect the issue would have been already spotted in such a case).